src/dstore/dstore_internal.h

```diff
@@ -28,6 +28,9 @@
 
 #include "dstore.h" /* import public data types */
 
+#define IO_DATA  0x001
+#define IO_NO_DATA    0x002
```

Comment:
Let's convert it into an enum -- they are easier to be used with a debugger.

Also, I guess it is enough to have only one flag. We cannot have them both set, so probably
it is enough to have only one.

```c
enum dstore_io_vec_flags {
    DSTORE_IVF_NO_IO_DATA = 1 << 1,
}

static bool dstore_io_vec_flags_has_data(uint64_t flags)
{
  return (flags & DSTORE_IVF_NO_DATA) == 0;
}
```

Then you can replace `io_vec->flags & IO_NO_DATA` with `!dstore_io_vec_flags_has_data(io_vec->flags)` and `io_vec->flags & IO_DATA` with `dstore_io_vec_flags_has_data(io_vec->flags)`. We don't have to use this particular name and we even don't have to introduce this function (it can be a define or just bare expression). The main idea here is to avoid ambiguity whenever it is possible.

If you would like to keep both flags then please add a condition to the invariant for the vector:

```c
bool consistent_flags = !(flags & IO_NO_DATA) == !!(flags & IO_DATA);
```

Reply:
I have changed the name of flag and added function told by you. There is only one flag now. But I could not make it enum.